docs/development.md

```diff
@@ -0,0 +1,222 @@
+# Development
```

Comment:
Perhaps defining the terms at the beginning is not so important as sometimes the ToC can contain undefined terms. However, it is still good to give an overview of the document at the beginning because now that we are fragmenting our docs to many different files, it's good to define the purpose of each document clearly.